manager/orchestrator/service.go

```diff
@@ -47,22 +47,27 @@ func SetServiceTasksRemove(ctx context.Context, s *store.MemoryStore, service *a
 	err = s.Batch(func(batch *store.Batch) error {
 		for _, t := range tasks {
 			err := batch.Update(func(tx store.Tx) error {
+				// the task may have changed for some reason in the meantime
+				// since we read it out, so we need to get from the store again
+				// within the boundaries of a transaction
+				latestTask := store.GetTask(tx, t.ID)
```

Comment:
Two questions:

1. I forget, but is it possible for the task to change from this point on until the update on line 70?

2. If the update fails because of out of sequence, would it be better to simply retry?

@dperny